src/v/cluster/tm_stm_types.h

```diff
@@ -15,16 +15,63 @@
 namespace cluster {
 
 enum tx_status : int32_t {
-    ongoing,
-    preparing,
-    prepared,
-    aborting, // abort is initiated by a client
-    killed,   // abort is initiated by a timeout
-    ready,
-    tombstone,
+    /**
+     * When transactional id to producer id mapping is added to coordinator it
+     * starts in this state
+     */
+    empty = 5,
+    /**
+     * Transaction is ongoing as soon as partition or offset is added to it
+     */
+    ongoing = 0,
+    /**
+     * Transaction is marked as preparing_commit when it is requested to be
+     * committed by the client but before committing transaction on data and
+     * offset partitions
+     */
+    preparing_commit = 1,
+    /**
+     * Transaction is completed commit when its participants committed the
+     * transaction. This state is not replicated as committing data on
```

Comment:
> This state is not replicated

I see finish_transaction() is replicating these, did I miss something or is this comment stale?